File: algorithms/2073.time-needed-to-buy-tickets/solution.rb
# @param {Integer[]} tickets
# @param {Integer} k
# @return {Integer}
def time_required_to_buy(tickets, k)
  seconds = 0

  while true
    for i in 0...tickets.size
      if tickets[i] > 0
        seconds += 1
        tickets[i] -= 1

        return seconds if i == k && tickets[i] == 0
      end
    end
  end

  seconds
end

File: algorithms/2073.time-needed-to-buy-tickets/solution.go
func timeRequiredToBuy(tickets []int, k int) int {
	seconds := 0

	for {
		for i := 0; i < len(tickets); i++ {
			if tickets[i] > 0 {
				seconds++
				tickets[i]--

				if k == i && tickets[i] == 0 {
					return seconds
				}
			}
		}
	}

	return seconds
}
